Travis tweaks

awemany opened the pull request.

awemany committed:
Travis: Only show last 100 lines of debug.logs

awemany committed:
rpc-tests: Add test name to temporary directory

sickpig commented on qa/rpc-tests/test_framework/test_framework.py:
```diff
@@ -142,7 +142,17 @@ def main(self,argsOverride=None,bitcoinConfDict=None,wallets=None):
+
+        from os.path import basename
+        import string
+        from sys import argv
```
I rather add this `import` statement on top of the file. 

Other than that changes seems ok from here

awemany replied:
Done.

gandrewstone commented on qa/rpc-tests/test_framework/test_framework.py:
```diff
@@ -142,7 +145,14 @@ def main(self,argsOverride=None,bitcoinConfDict=None,wallets=None):
+
+        default_tempdir = tempfile.mkdtemp(prefix="test_"+testname+"_")
+
+        parser.add_option("--tmpdir", dest="tmpdir", default=default_tempdir,
```
The QA tests used to have a very convenient workflow because it laid run 1 right on top of run 2, to [tmpdir]/node0, for example.  This meant that you could very quickly reload various output files, etc to see the next run.  

A few months ago we committed something that broke that since it now wrote to [tmpdir]/[pid].  Now it looks like this change proposes [tmpdir]/[pid]/test_[testname].

Presumably this is to stop different tests from overwriting each other when running pull-tester.py?

Perhaps its time to bifurcate this flag and have a "--tmppfx" (temp directory prefix) that does what you've defined here, and a "--tmpdir" that restores the old way and simpler debugging workflow?

awemany replied:
Hmm, I am not sure I understand the issue and your goals correctly yet :)

Before this change, the runs still used `mkdtemp()` to make the directory, which should be unique for every run. This is also necessary in case you run on a machine that runs multiple regtests at once (I do this on the `buildbot` host).

Are you saying that in an even earlier code state, the temporary directories were not really random and always put the data into the same one?

And you want `--tmpdir` to set the output directory and `--tmppfx` to set a prefix that is then combined with a random run ID a.k.a the current `mkdtemp()` behavior?

What do you mean when you say that 'it laid run 1 right on top of run 2'? I can imagine that run 2 appends the debug.log of run 1, but what happens with a configuration change for example? That would overwrite the `bitcoin.conf`, wouldn't it.

EDIT: I just searched the code and the `mkdtemp()` approach is in there since a long time, I found this commit by Gavin Andresen from 2014, for example: 

e8097f7df164b4bf799963e5ab2539c36079187d

gandrewstone replied:
yes, if you specified a tmpdir as /tmp/test then it would put your run at /tmp/test, not /tmp/test/[pid] as happens now.  I would love it if this functionality was re-enabled because is makes debugging qa runs much easier.  For example rather than just doing ./bitcoin-cli -conf=/tmp/test/node0 getinfo (or more likely a few up arrow keys to use command line recall), I have to do:
"ps -efww | grep bitcoind"
[find the correct bitcoind instance, and find its parent (the .py file) pid]
./bitcoin-cli -conf=/tmp/test/[pid]/node0 getinfo

Loading debug.log into an editor or tail -f ing it is similarly painful.

WRT Gavin's commit (although he didn't do this in that commit, just moved it around -- which is good because the code seems to create the temp dir even if unused):
parser.add_option("--tmpdir", dest="tmpdir", default=tempfile.mkdtemp(prefix="test"),
                          help="Root directory for datadirs")

You can see that the mkdtemp is only used if tmpdir is not defined.

awemany committed:
Change --tmpdir to --tmppfx

As per review by @gandrewstone on PR #1250. Also document current
behavior as setting a prefix (and not the data directory itself).

awemany committed:
Add tmpdir option to explicitly define the temporary directory

The --tmpdir option overrides the --tmppfx option if it is specified.
